### PR TITLE
Add tests for chunking module

### PR DESCRIPTION
**Pull request by violenil**
This PR adds some more tests to the chunking methods, and adds back the semantic chunking logic.

### CODE REVIEW

File: chunked_pooling/chunking.py [hunk 1]
```diff
@@ -56,27 +56,38 @@ def __init__(
                 trust_remote_code=True,
             )
         else:
-            self.tokenizer = tokenizer
+            self.tokenizer = tokenizer or AutoTokenizer.from_pretrained("bert-base-uncased")
```
Comment (guenthermi) [inline chunked_pooling/chunking.py]: I think we should not initialize a bert tokenizer by default, this can lead to unexpected errors

**Review by guenthermi — COMMENTED** [review]
The tests look already good, but it would be nice to have something more explicitly. For example some text snippets and the expected boundaries.

For the modifications to the chunking. Probably we need to clean this up a bit further, I want to get rid of all the default values

File: chunked_pooling/chunking.py [hunk 2]
```diff
@@ -19,13 +19,74 @@ def __init__(
...
+                [Document(text=text)], show_progress=False
+            )
+        ]
+
+        # Tokenize the entire text
+        tokens = tokenizer.encode_plus(
+            text,
+            return_offsets_mapping=True,
+            add_special_tokens=False,
+            max_length=512,
```
Comment (guenthermi) [inline chunked_pooling/chunking.py]: is this correct to set max-length=512?

File: run_chunked_eval.py [hunk 3]
```diff
@@ -104,7 +99,6 @@ def main(model_name, strategy, task_name, eval_split):
         output_folder='results-normal-pooling',
         eval_splits=[eval_split],
         overwrite_results=True,
-        batch_size=BATCH_SIZE,
```
Comment (guenthermi) [inline run_chunked_eval.py]: I think this should not be remove as here is another batch size argument:
https://github.com/jina-ai/late-chunking/blob/8237bf2c36d46e10d93220d6ee904d7d53210f83/chunked_pooling/mteb_chunked_eval.py#L105
I know it is a bit messy ...

**Review by guenthermi — APPROVED** [review]
I found two small things other than that it looks good

File: chunked_pooling/chunking.py (hunk 2 above)
Comment (violenil) [reply chunked_pooling/chunking.py]: I can probably remove this.

File: run_chunked_eval.py (hunk 3 above)
Comment (violenil) [reply run_chunked_eval.py]: that's annoying, but OK.

**Pull request merged**